src/rpc/rpc.cc

```diff
@@ -1184,6 +1184,19 @@ future<> server::connection::send_unknown_verb_reply(std::optional<rpc_clock_typ
       ).discard_result();
   }
 
+  void server::abort_connection(connection_id id) {
+      auto it = _conns.find(id);
+      if (it == _conns.end()) {
+          return;
+      }
+      try {
+          it->second->abort();
+      } catch (...) {
+          log_exception(*it->second, log_level::error,
+                        "fail to shutdown connection on user request", std::current_exception());
+      }
+  }
```

Comment:
I can see future uses wanting to know when the connection was drained, but no need to implement that now.